### ISSUE TITLE
[DRC] edge-to-edge (singluar point of connection) metals doesn't trigger any violation

### ISSUE THREAD

**Issue by tskaar** [aesc-silicon/gdscheck#13]
### Process / PDK

ihp-sg13g2

### gdscheck version

gdscheck 0.1.2

### Top cell

TOP

### Command line

```shell
gdscheck run -p ihp-sg13g2 -i edge2edge.gds -t TOP --suite core --report gdscheck_core.lyp
```

### What's wrong?

False negative (a real violation is not reported)

### Rule ID(s)

Mn.a, Mn.b

### Description

When having two metals (might be applicable to all layers that have minimum dimension/spacing?) have a single point of contact, one does not trigger the minimum sizing rule and/or minimum spacing.

A typical trigger case for this would be something like this:

<img width="999" height="1002" alt="Image" src="https://github.com/user-attachments/assets/b21ae24d-83ed-426a-b48d-f0370e01ebd7" />

Where there should be a violation at the crossing in the middle here.
In this particular case, the KLayout DRC deck triggers both M1.a (minimum width of 0.16 µm) and M1.b (minimum space of 0.18 µm)

### GDS file and marker database

[edge_to_edge.zip](https://github.com/user-attachments/files/32190417/edge_to_edge.zip)

Zipped is:

```
edge_to_edge/
  edge2edge.gds --- GDS file in use
  drc_run_TOP/ --- Results from KLayout DRC deck
  gdscheck_core.lyp --- Results from gdscheck using Core suite.
```

### Confirmations

- [x] I have the right to share this layout publicly (this is a public repository under AGPL-3.0-or-later).
- [x] I tried to reduce the GDS to the smallest reproducer I could.

### Environment

OS: ArchLinux x86_64

Compiled using:
```
export RUSTUP_TOOLCHAIN=stable
export CARGO_TARGET_DIR=target
cargo build --frozen --release --all-features
```

Cargo version: `cargo 1.98.0 (797e8a9bc 2026-08-05)`
Rustc version: `rustc 1.98.0 (88d9e12ae 2026-08-18)`

**Comment by dnltz** [comment]
Thanks for reporting this issue @tskaar - I verified it with #14 and this version reported both violations:

```
Topcell: TOP
DRC completed in 0.041s
2 violation(s):
  [M1.a] 1
  [M1.b] 1
```

So, should be fixed soon. I also added a test pattern to each metal layer (for IHP) to make sure this won't be silent in the future.
